### PR TITLE
reset-state: only restart snapd if it was running. Also, start/stop the socket.

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: reset-state [hunk 1]
```diff
@@ -60,8 +65,10 @@ set -x
...
-echo
-echo "Starting snapd"
-set -x
-systemctl start snapd
-set +x
+if [ "$snapd_was_active" = "yes" ]; then
+    echo
+    echo "Starting snapd"
+    set -x
+    systemctl start snapd.socket
```
Comment (zyga) [inline reset-state:72]: what about .service?

**Comment by zyga** [comment]
Thanks a lot! Just one question inline.

File: reset-state (hunk 1 above)
Comment (chipaca) [inline reset-state:72]: it's started automagically by systemd when the socket is poked

Comment (chipaca) [inline reset-state:72]: note this means that stopping just `snapd` is pointless, as systemd will happily start it again as soon as the socket is accessed. If it is running when you do `systemctl stop snapd` it'll print a warning about this.

Comment (chipaca) [inline reset-state:72]: (`snapd` is short for `snapd.service`; I spell it out in the script to make it clearer)

**Pull request merged**